backends/ent/retrieve.go

```diff
@@ -28,50 +34,109 @@ func (backend *Backend) Retrieve(id string, _ *storage.RetrieveOptions) (*sbom.D
 		backend.Options = NewBackendOptions()
 	}
 
-	entDoc, err := backend.client.Document.Query().
-		Where(document.HasMetadataWith(metadata.IDEQ(id))).
-		WithMetadata().
-		WithNodeList().
-		Only(backend.ctx)
+	documents, err := backend.GetDocumentsByID(id)
 	if err != nil {
-		return nil, fmt.Errorf("eager loading document edges: %w", err)
+		return nil, fmt.Errorf("querying documents: %w", err)
 	}
 
-	entDoc.Edges.Metadata, err = entDoc.QueryMetadata().
-		WithAuthors().
-		WithDocumentTypes().
-		WithTools().
-		Only(backend.ctx)
-	if err != nil {
-		return nil, fmt.Errorf("eager loading metadata edges: %w", err)
+	if len(documents) > 1 {
+		return nil, fmt.Errorf("%w %s", errMultipleDocuments, id)
+	}
+
+	return documents[0], nil
+}
+
+func (backend *Backend) GetDocumentsByID(ids ...string) ([]*sbom.Document, error) {
```

Comment:
I'm still trying to think how to handle the case where documents have the same ID. Somehow the list dumping them all does not seem that useful. Once we find the right pattern we can promote it to the official storage interface.